### PR TITLE
gap 4.13.0 (new formula)

### PR DESCRIPTION
**Pull request by bias**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by bias** [comment]
Bumping this, not sure how to request review?

File: Formula/g/gap.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  desc "Groups, Algorithms, Programming - A system for computational discrete algebra"
+  homepage "https://www.gap-system.org"
+  url "https://github.com/gap-system/gap/releases/download/v4.13.0/gap-4.13.0.tar.gz"
+  sha256 "cc76ecbe33d6719450a593e613fb87e9e4247faa876f632dd0f97c398f92265d"
+  license "GPL-2.0-or-later"
+
+  depends_on "autoconf" => :build
+  depends_on "libtool" => :build
+  depends_on "gmp"
+  depends_on :macos
```
Comment (SMillerDev) [inline Formula/g/gap.rb:11]: Why does it specifically depend on macOS?

File: Formula/g/gap.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  depends_on "readline"
+  depends_on :xcode
+  uses_from_macos "zlib"
+
+  def install
+    system "./configure",
+      "--prefix=#{prefix}",
+      "--with-gmp=#{Formula["gmp"].include}",
+      "--with-readline=#{Formula["readline"].include}"
```
Comment (SMillerDev) [inline Formula/g/gap.rb:20]:
```suggestion
    system "./configure", *std_configure_args,
      "--with-gmp=#{Formula["gmp"].include}",
      "--with-readline=#{Formula["readline"].include}"
```

File: Formula/g/gap.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    system "make", "install"
+    pkgshare.install "pkg"
```
Comment (SMillerDev) [inline Formula/g/gap.rb:22]: What is this for?

File: Formula/g/gap.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  end
+
+  test do
+    system bin/"gap", "--version"
```
Comment (SMillerDev) [inline Formula/g/gap.rb:26]: If we do a version check, let's see if it works:
```suggestion
    assert_match version.to_s, shell_output("#{bin}/gap --version")
```

File: Formula/g/gap.rb [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+    (testpath/"m11.gap").write <<-EOS
+      m11 := Group((1,2,3,4,5,6,7,8,9,10,11),(3,7,11,8)(4,10,5,6));
+      s := Size( m11 );
+      PrintTo("m11.txt", s);
+      QUIT;
+    EOS
+    system bin/"gap", "m11.gap"
+
+    assert_equal "7920", `cat m11.txt`.strip
```
Comment (SMillerDev) [inline Formula/g/gap.rb:36]: Can we use `shell_output` to check the output here instead of using backticks?

File: Formula/g/gap.rb (hunk 3 above)
Comment (bias) [reply Formula/g/gap.rb:22]: 1. For whatever reason gap tries to use the system gap and readline so they need to be specified.
2. GAP needs to run make install in order to use prefix etc
3. (presumably this is what you meant) GAP builds packages in the pkg folder but doesn't "install" them with make install so they need to be manually installed into pkgshare

File: Formula/g/gap.rb (hunk 1 above)
Comment (bias) [reply Formula/g/gap.rb:11]: GAP has a fairly different build for Mac vs Linux. I'm developing on Mac so only did the macOS option. I could try setting up separate linux / Mac dependencies and builds if that's necessary - I figured that could be added later by myself or someone else.

File: Formula/g/gap.rb (hunk 4 above)
Comment (bias) [reply Formula/g/gap.rb]: Removed version check. The functionality test is valid and the tar has a hashsum - not really an important test.

File: Formula/g/gap.rb (hunk 5 above)
Comment (bias) [reply Formula/g/gap.rb]: OK, modified that. I was going to try an pipe from GAP but writing to a file works, using cat in a shell probably isn't the most ideal either but works.

File: Formula/g/gap.rb (hunk 1 above)
Comment (SMillerDev) [reply Formula/g/gap.rb:11]: Let's try and build on Linux too, since there is nothing technically blocking it.

File: Formula/g/gap.rb (hunk 3 above)
Comment (SMillerDev) [reply Formula/g/gap.rb:21]: Yeah, only meant 3 but GitHub quotes some earlier lines too. Can you ask upstream to rectify this missing part of `make install`?

File: Formula/g/gap.rb (hunk 1 above)
Comment (bias) [reply Formula/g/gap.rb:11]: OK, I do have 2 Linux servers I can dev this on - I'll add Home brew to the Ubuntu box and get this working. Sorry for the delay busy lately.

File: Formula/g/gap.rb (hunk 3 above)
Comment (bias) [reply Formula/g/gap.rb:21]: I can ask them but I doubt they'll change that unless I take over the work. I will mention this to them.

File: Formula/g/gap.rb (hunk 1 above)
Comment (bias) [reply Formula/g/gap.rb]: I have it working on my dev machine (Mac) and linux server. Not quite sure how ideally I've dealt with the dependencies: for Mac and Linux I added a 'gcc' and 'make' dependency ... the tool will however use any that it finds in the system path - that dependency will ensure I can run based on my understanding of home-brew gcc. However, this doesn't force the usage of hombrews gcc.

**Comment by chenrui333** [comment]
@bias you have to squash all the commits and forced update your PR, the merge commit approach wont work for homebrew-core PRs.

**Pull request closed without merging**